test/integration/puppetlabs/services/jruby/jruby_puppet_internal_test.clj

```diff
@@ -1,35 +1,60 @@
 (ns puppetlabs.services.jruby.jruby-puppet-internal-test
   (:require [clojure.test :refer :all]
             [puppetlabs.services.jruby.jruby-puppet-internal :as jruby-internal]
-            [puppetlabs.services.jruby.jruby-testutils :as jruby-testutils])
-  (:import (com.puppetlabs.puppetserver.pool JRubyPool)))
+            [puppetlabs.services.jruby.jruby-testutils :as jruby-testutils]
+            [puppetlabs.services.jruby.jruby-puppet-schemas :as jruby-schemas])
+  (:import (com.puppetlabs.puppetserver.pool JRubyPool)
+           (org.jruby RubyInstanceConfig$CompileMode)
+           (clojure.lang ExceptionInfo)))
 
-  (deftest ^:integration settings-plumbed-into-jruby-container
-    (testing "setting plumbed into jruby container for"
-      (let [pool (JRubyPool. 1)
-            config (jruby-testutils/jruby-puppet-config
-                     {:http-client-connect-timeout-milliseconds 2
-                      :http-client-idle-timeout-milliseconds 5
-                      :http-client-cipher-suites ["TLS_RSA_WITH_AES_256_CBC_SHA256"
-                                                  "TLS_RSA_WITH_AES_256_CBC_SHA"]
-                      :http-client-ssl-protocols ["TLSv1" "TLSv1.2"]})
-            instance (jruby-internal/create-pool-instance! pool 0 config #() nil)
-            container (:scripting-container instance)]
-        (try
-          (let [settings (into {} (.runScriptlet container
-                                    "java.util.HashMap.new
-                                       (Puppet::Server::HttpClient.settings)"))]
-            (testing "http_connect_timeout_milliseconds"
-              (is (= 2 (settings "http_connect_timeout_milliseconds"))))
-            (testing "http_idle_timeout_milliseconds"
-              (is (= 5 (settings "http_idle_timeout_milliseconds"))))
-            (testing "cipher_suites"
-              (is (= ["TLS_RSA_WITH_AES_256_CBC_SHA256"
-                      "TLS_RSA_WITH_AES_256_CBC_SHA"]
-                    (into [] (settings "cipher_suites")))))
-            (testing "ssl_protocols"
-              (is (= ["TLSv1" "TLSv1.2"]
-                    (into [] (settings "ssl_protocols"))))))
-          (finally
-            (.terminate (:jruby-puppet instance))
-            (.terminate container))))))
+(deftest get-compile-mode-test
```

Comment:
I totally agree with this and had the same confusion / concern when writing this.

Where I ended up landing on it was:

1. I kind of hate having multiple files with the same name in a project, which has been a factor in how we've chosen to name a lot of stuff so far, and I was hesitant to create another `jruby_puppet_internal_test` namespace.
2. I don't feel like we have solid alignment / agreement on the best way to organize unit vs. integration tests as a team, and have been thinking for a while that that would be a good topic for a festivus sometime soon.  So I made a somewhat arbitrary decision in the meantime.

I did have the same thought though, and am willing to change it up if anyone has a brilliant suggestion that is actionable prior to us having a team conversation about best practices.